### ISSUE TITLE
Store/vuex can not find mutation in sub module

### ISSUE THREAD

**Issue by larsmars** [nathantsoi/vue-native-websocket#40]
```
import VueNativeSock from 'vue-native-websocket'
Vue.use(VueNativeSock, 'ws://localhost:3000', {store: store, format: 'json'})
```
vuex.esm.js?358c:383 [vuex] unknown mutation type: SOCKET_ONMESSAGE

this mutations is defined in the sub module **socket**:
/socket/mutations.js

```
  SOCKET_ONMESSAGE (state, message) {
    for (var property in message) {
      console.log(property + ' = ' + message[property])
    }
  }
```
In index.js:

```
import socketModule from '@/store/modules/socket

Vue.use(Vuex)

export const store = new Vuex.Store({
  modules: {
    socket: socketModule
  },
```

If the mutations is defined in index.js everything works

**Comment by larsmars** [comment]
I understand this is commited:

SOCKET_ONOPEN
SOCKET_ONCLOSE
SOCKET_ONERROR
SOCKET_ONMESSAGE

It look`s like vuex doesn`t map this to the sub module.
 
Is there som way to map this to dispatch action? or other solutions?
ie: 
`store.dispatch('socket/onmessage', message)`

**Comment by ejwaibel** [comment]
+1 Any chance this will be fixed anytime soon?

**Comment by robben1234** [comment]
I'm solving it with redirecting eventName (adding 'moduleName/' to eventName) in passToStoreHandler. But in current version it's not possible to offload commit to original handler, you need some edits to make it work:

1. `if (!eventName.includes('SOCKET_')) {` instead of startsWith as you are redirecting to another module.

2. `eventName` should be uppercase() even before you make your redirect. Then for target you should your redirected eventName.

**Comment by UglyHobbitFeet** [comment]
Yeah, I had to do a similar hack. I took the passToStoreHandler method shown in the readme, and changed let target = eventName.toUpperCase() into let target = 'myStorePath/' + eventName.toUpperCase(). Far from ideal, but it works for my particular scenario.

**Comment by fjxhkj** [comment]
My solution:
```js
Vue.use(VueNativeSock, 'ws://api.test.com:666', {
  store: store,
  passToStoreHandler: function (eventName, event) {
    if (!eventName.startsWith('SOCKET_')) { return }
    let method = 'commit'
    let target = eventName.toUpperCase()
    let msg = event
    if (this.format === 'json' && event.data) {
      msg = JSON.parse(event.data)
      if (msg.mutation) {
        target = [msg.namespace || '', msg.mutation].filter((e) => !!e).join('/')
      } else if (msg.action) {
        method = 'dispatch'
        target = [msg.namespace || '', msg.action].filter((e) => !!e).join('/')
      }
    }
    // Write this if multiple stores are used and namespace options are enabled
    target = "store/" + target
    this.store[method](target, msg)
  }
});
```

**Comment by ejwaibel** [comment]
Hacking the call to VueNativeSock doesn't seem like a solution to the actual problem. For those of you with a "solution" are you able to fix the underlying code and submit a PR to finally fix this issue?

**Comment by hcq11** [comment]
> My solution:
> 
> ```js
> // If your `user` namespaced store is here
> import storeUser from 'store/user'
> 
> Vue.use(VueNativeSock, 'ws://api.test.com:666', {
>   store: storeUser,
>   passToStoreHandler: function (eventName, event) {
>     if (!eventName.startsWith('SOCKET_')) { return }
>     let method = 'commit'
>     let target = eventName.toUpperCase()
>     let msg = event
>     if (this.format === 'json' && event.data) {
>       msg = JSON.parse(event.data)
>       if (msg.mutation) {
>         target = [msg.namespace || '', msg.mutation].filter((e) => !!e).join('/')
>       } else if (msg.action) {
>         method = 'dispatch'
>         target = [msg.namespace || '', msg.action].filter((e) => !!e).join('/')
>       }
>     }
>     // Write this if multiple stores are used and namespace options are enabled
>     target = "user/" + target
>     this.store[method](target, msg)
>   }
> });
> ```

work for me

**Comment by skadiD** [comment]
> 我的解决方案：
> 
> ```js
> //如果您的`user`命名空间存储在此处，
> 则从'store / user'导入 storeUser  
> 
> Vue 。使用（VueNativeSock ， 'WS：//api.test.com：666' ， {
>   店：storeUser ，
>   passToStoreHandler：功能 （eventName的， 事件） {
>     如果 （！eventName的。startsWith （'SOCKET_' ）） {返回}让方法= '提交'让目标= eventName的。toUpperCase （）让MSG   
>        
>        
>       = 事件
>     if  （此。格式 ===  'json'  && 事件。数据） { 
>       msg  =  JSON 。解析（事件。数据）
>       如果 （MSG 。突变） {
>         目标 =  [ MSG 。命名空间 ||  '' ， 味精。突变] 。滤波器（（ë ） => ！ ！ Ë）。加入（'/' ）
>       } 否则 如果 （MSG 。动作） {
>         方法 =  '调度'
>         目标 =  [ MSG 。命名空间 ||  '' ， 味精。动作] 。滤波器（（ë ） => ！ ！ ë ）。加入（“/” ）} } //如果使用多个存储写入这和命名空间选项启用目标
>       
>     
>     
>      =  “用户/”  + 定位
>     到此。存储[方法] （target ， msg ）
>   } 
> } ）;
> ```



> My solution:
> 
> ```js
> // If your `user` namespaced store is here
> import storeUser from 'store/user'
> 
> Vue.use(VueNativeSock, 'ws://api.test.com:666', {
>   store: storeUser,
>   passToStoreHandler: function (eventName, event) {
>     if (!eventName.startsWith('SOCKET_')) { return }
>     let method = 'commit'
>     let target = eventName.toUpperCase()
>     let msg = event
>     if (this.format === 'json' && event.data) {
>       msg = JSON.parse(event.data)
>       if (msg.mutation) {
>         target = [msg.namespace || '', msg.mutation].filter((e) => !!e).join('/')
>       } else if (msg.action) {
>         method = 'dispatch'
>         target = [msg.namespace || '', msg.action].filter((e) => !!e).join('/')
>       }
>     }
>     // Write this if multiple stores are used and namespace options are enabled
>     target = "user/" + target
>     this.store[method](target, msg)
>   }
> });
> ```

work for me and could you please submit PR to help us
